build-system/tasks/helpers.js

```diff
@@ -392,7 +392,12 @@ function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
 
   if (options.watch) {
     bundler = watchify(bundler);
-    bundler.on('update', () => performBundle(/* continueOnError */ true));
+    bundler.on('update', () => {
+      const bundleComplete = performBundle(/* continueOnError */ true);
+      if (options.onWatchBuild) {
+        options.onWatchBuild(bundleComplete);
+      }
+    });
```

Comment:
This mechanism will only work for non-extension files. We use a different one for extensions (that watches the entire directory). That code lives here:

https://github.com/ampproject/amphtml/blob/5e6a4395f4bf8dda25d035d8d7d3c2556c6d4e34/build-system/tasks/extension-helpers.js#L438-L452

I think you'd have to add similar code there too.

Reply:
Oh yes! I was reviewing on mobile, didn't see the other change.